astropy/units/quantity.py

```diff
@@ -481,13 +481,15 @@ def __new__(cls, value, unit=None, dtype=np.inexact, copy=True, order=None,
                     # structured dtype of the value.
                     dtype = unit._recursively_get_dtype(value)
 
+        USING_DEFAULT_UNIT = False
```

Comment:
Rereading this, I wonder why you use all-caps here. I tended to treat those as "environment variables", i.e., the caps indicate that the variable is defined outside of the scope. But that is not the case here, so maybe one should use lower case?

Reply:
SGTM. I'll repush and merge on pass.